experimental/dds/tree2/src/test/rebase/rebaseBranch.spec.ts

```diff
@@ -0,0 +1,202 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert, fail } from "assert";
+import { validateAssertionError } from "@fluidframework/test-runtime-utils";
+// Allow importing from these specific files which are being tested:
+/* eslint-disable-next-line import/no-internal-modules */
+import { GraphCommit, RevisionTag, findCommonAncestor, rebaseBranch } from "../../core/rebase";
+import { NonEmptyTestChange, TestChange, TestChangeRebaser } from "../testChange";
+
+function node(
+	inputContext: readonly number[],
+	intention: number | number[],
+	parent?: GraphCommit<TestChange>,
+): GraphCommit<TestChange> {
+	return {
+		change: TestChange.mint(inputContext, intention),
```

Comment:
Can we derive the input context from the parent commit instead of passing it in as an argument?

Reply:
I think so - good idea. I'm going to make a small work item and do that in a followup, because it's a little annoying to do with the current `findAncestor` helpers and those can be tweaked too.